client/index.js

```diff
@@ -39,7 +39,10 @@ __webpack_public_path__ = `${assetPrefix}/_next/webpack/` //eslint-disable-line
 // Initialize next/asset with the assetPrefix
 asset.setAssetPrefix(assetPrefix)
 // Initialize next/config with the environment configuration
-envConfig.setConfig(runtimeConfig || {})
+envConfig.setConfig({
+  serverRuntimeConfig: {},
+  publicRuntimeConfig: runtimeConfig
```

Comment:
This is interesting part :)

Do you think it makes sense to keep `serverRuntimeConfig` as an empty object on the client, rather than `null` or something like that? I see some benefit in throwing at any reference to it, instead of failing quietly.

What do you think?

Reply:
Hmmm to me it makes sense to set it as empty object, as that was the behaviour already, and it'd help when destructuring eg:

```
import getConfig from 'next/config'
const {serverRuntimeConfig, publicRuntimeConfig} = getConfig()
const {something} = serverRuntimeConfig
```

In theory you'd be able to do this using default assignment though (`= {}`):

```
import getConfig from 'next/config'
const {serverRuntimeConfig = {}, publicRuntimeConfig} = getConfig()
const {something} = serverRuntimeConfig
```

Reply:
I see. I'm fine with either way, just wanted to bring this up. 
Thanks for the fix.